### PR TITLE
[MRG] Add example to "a quick guide for testing"

### PR DESCRIPTION
**Pull request by standage**
This PR fulfills a suggestion from @bocajnotnef in #1243. The testing docs now have a simple example to show basic testing mechanics.

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by standage** [comment]
Ready for review and merge @betatim @ctb @camillescott @luizirber.

**Comment by betatim** [comment]
Should we add a test or explanation of existing one that loads data from `test-data/`?

Maybe all that is needed is to point out that there is some helper infrastructure available for getting data and running scripts that take care of cleanup etc.

**Comment by standage** [comment]
Yea, that's a good idea. I approached this PR as a generic (but relevant) example for testing in Python, but it would be good to also highlight some of the specifics to testing in khmer.

File: doc/dev/a-quick-guide-to-testing.rst [hunk 1]
```diff
@@ -139,11 +139,23 @@ review.
```
Comment (betatim) [inline doc/dev/a-quick-guide-to-testing.rst]: To test the command line interface of a script take a look at `tests/test_scripts.py` to learn about the helper functions that exist for running scripts, capturing their output, and tidying up afterwards.

There are helper functions for loading data and generating temporary names in `khmer_tst_utils.py`.

---

Something like this is what I was thinking of for my comment.

Comment (standage) [inline doc/dev/a-quick-guide-to-testing.rst]: Great, thanks.

**Comment by betatim** [comment]
:+1:

**Pull request merged**